src/sentry/api/serializers/models/plugin.py

```diff
@@ -77,18 +90,9 @@ def serialize(self, obj, attrs, user):
         if obj.author:
             d["author"] = {"name": str(obj.author), "url": str(obj.author_url)}
 
-        # TODO(Leander): Convert this field to:
-        # d["isDeprecated"] = obj.slug in SHADOW_DEPRECATED_PLUGINS or datetime.today() > deprecation_date
-        # but we are late on deprecating right now
-        d["isDeprecated"] = obj.slug in SHADOW_DEPRECATED_PLUGINS
-
-        d["isHidden"] = (
-            not features.has(
-                SHADOW_DEPRECATED_PLUGINS.get(obj.slug), getattr(self.project, "organization", None)
-            )
-            if d["isDeprecated"]
-            else d.get("enabled", False) is False and obj.is_hidden()
-        )
+        d["isDeprecated"] = is_plugin_deprecated(obj, self.project)
+
+        d["isHidden"] = d["isDeprecated"] or d.get("enabled", False) is False and obj.is_hidden()
```

Comment:
There are two boolean operators on this line so you might want to add parens for clarity. I'd also just use `not` instead of `is False`:
```py
d["isHidden"] = d["isDeprecated"] or (not d.get("enabled", False) and obj.is_hidden())
```

Reply:
good catch! implemented